controller/lib/core/controller.cpp

```diff
@@ -19,65 +19,69 @@ limitations under the License.
 #include "vars.h"
 #include <math.h>
 
-// PID-tuning were chosen by following the Ziegler-Nichols method,
-// https://en.wikipedia.org/wiki/Ziegler%E2%80%93Nichols_method
-//
-// Note that Ku and Tu only seem to work with this particular sample time.
 static constexpr Duration PID_SAMPLE_PERIOD = milliseconds(10);
-static constexpr float Ku = 200;
-static constexpr Duration Tu = seconds(1.5f);
-
-// "No overshoot" settings from the Ziegler-Nichols Wikipedia page.  This
-// avoids overpressurizing the patient's lungs.
-static constexpr float Kp = 0.2f * Ku;
-static constexpr float Ki = 0.4f * Ku / Tu.seconds();
-static constexpr float Kd = Ku * Tu.seconds() / 15;
 
 static DebugFloat dbg_setpoint("setpoint", "Setpoint pressure, kPa");
 
+static DebugFloat dbg_kp("kp", "Proportional gain for main loop", 3.5);
+static DebugFloat dbg_ki("ki", "Integral gain for main loop", 1.0);
+static DebugFloat dbg_kd("kd", "Derivitive gain for main loop");
+static DebugFloat dbg_sp("pc_setpoint", "Pressure control setpoint (cmH2O)");
+
 Controller::Controller()
-    : pid_(Kp, Ki, Kd, ProportionalTerm::ON_ERROR,
-           DifferentialTerm::ON_MEASUREMENT,
-           // Increases in the blower fan speed should result in increased
-           // pressure.
-           ControlDirection::DIRECT,
-           // Our output is an 8-bit PWM.
-           /*output_min=*/0.f, /*output_max=*/255.f, PID_SAMPLE_PERIOD) {}
+    : pid_(0.0f, 0.0f, 0.0f, ProportionalTerm::ON_ERROR,
+           DifferentialTerm::ON_MEASUREMENT, 0.f, 1.0f, PID_SAMPLE_PERIOD) {}
 
 Duration Controller::GetLoopPeriod() { return PID_SAMPLE_PERIOD; }
 
 ActuatorsState Controller::Run(Time now, const VentParams &params,
                                const SensorReadings &readings) {
   BlowerSystemState desired_state = fsm_.DesiredState(now, params);
 
-  return {.fan_setpoint_cm_h2o = desired_state.setpoint_pressure.cmH2O(),
-          .expire_valve_state = desired_state.expire_valve_state,
-          .fan_power = ComputeFanPower(now, desired_state, readings),
-          .fan_valve = 1.0f /* not used yet */};
-}
+  ActuatorsState actuator_state;
```

Comment:
Please add the following TODO (this is just the feedback I gave on Slack).

```
TODO(jlebar): Add a boolean in ActuatorState that indicates whether the blower should be on.  Then remove this switch statement; we shouldn't be switching on the VentMode here.
```

Reply:
I'll add that TODO, but I think we should have a discussion about this in slack.
In my opinion, ActuatorState should just hold the state of the actuators, it currently also has the pressure setpoint which doesn't really belong there.  Also, the structure being returned from the blower FSM needs some more though.  I'll move that discussion to slack though.